simpleviewinc/tap-events-force
# Zen 377 display props

**Ticket**: [issue_id](https://jira.simpleviewtools.com/browse/ZEN-377)

`[semver]: minor`

## Context

* the interface and examples that evf provided in the beginning of this project is updating slowly based on the back and forth we have in the slack channel.
they added a new interface `EFDisplayProperties` which contains:
`displayProperties: {     dateFormat: "MM/dd/yyyy",     timeFormat: EfAgendaTimeFormat.Twelve  }`
* use the `timeFormat` prop to determine whether or not we display our session times in 24hr or 12 hr format

## Goal

* Dynamically set the militaryTime setting based on the `displayProperties.timeFormat` prop


## Updates

* `src/actions/session/mapSessionInterface.js`
    * added `displayProperties` prop to `settings.displayProperties` store
* `src/actions/session/setAgendaSessions.js`
   * set the militaryTime bool based on our store data
* `src/components/sessions.js`
   * ensure we pass in the militaryTime boolean
* `src/mocks/eventsforce/testData.js`
   * added testdata for `displayProperties
## Testing

* Run `keg dpg run package=docker.pkg.github.com/simpleviewinc/keg-packages/tap:zen-377-display-props`
* navigate to http://local.kegdev.xyz/
* open the `Test Data (JSON)` editor
* change the `displayProperties.timeFormat` to either `"12"` or `"24"`
* hit apply
* verify that the timeblocks and session times update accordingly to use military time or not
* verify the session details modal reflect this change too
